app/src/main/java/com/zeapo/pwdstore/ui/dialogs/PasswordGeneratorDialogFragment.kt

```diff
@@ -0,0 +1,104 @@
+
+/** A placeholder fragment containing a simple view.  */
+class PasswordGeneratorDialogFragment : DialogFragment() {
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = MaterialAlertDialogBuilder(requireContext())
+        val callingActivity: Activity = requireActivity()
+        val inflater = callingActivity.layoutInflater
+        val view = inflater.inflate(R.layout.fragment_pwgen, null)
+        val monoTypeface = Typeface.createFromAsset(callingActivity.assets, "fonts/sourcecodepro.ttf")
+        builder.setView(view)
+        val prefs = requireActivity().applicationContext
+            .getSharedPreferences("PasswordGenerator", Context.MODE_PRIVATE)
+
+        view.findViewById<CheckBox>(R.id.numerals)?.isChecked = !prefs.getBoolean("0", false)
+        view.findViewById<CheckBox>(R.id.symbols)?.isChecked = prefs.getBoolean("y", false)
+        view.findViewById<CheckBox>(R.id.uppercase)?.isChecked = !prefs.getBoolean("A", false)
+        view.findViewById<CheckBox>(R.id.lowercase)?.isChecked = !prefs.getBoolean("L", false)
+        view.findViewById<CheckBox>(R.id.ambiguous)?.isChecked = !prefs.getBoolean("B", false)
+        view.findViewById<CheckBox>(R.id.pronounceable)?.isChecked = !prefs.getBoolean("s", true)
+
+        val textView: AppCompatEditText = view.findViewById(R.id.lengthNumber)
+        textView.setText(prefs.getInt("length", 20).toString())
+        val passwordText: AppCompatTextView = view.findViewById(R.id.passwordText)
+        passwordText.typeface = monoTypeface
+        builder.setPositiveButton(resources.getString(R.string.dialog_ok)) { _, _ ->
+            val edit = callingActivity.findViewById<EditText>(R.id.crypto_password_edit)
+            edit.setText(passwordText.text)
+        }
+        builder.setNegativeButton(resources.getString(R.string.dialog_cancel)) { _, _ -> }
+        builder.setNeutralButton(resources.getString(R.string.pwgen_generate), null)
+        val dialog = builder.setTitle(this.resources.getString(R.string.pwgen_title)).create()
+        dialog.setOnShowListener {
+            setPreferences()
+            try {
+                passwordText.text = generate(requireActivity().applicationContext)[0]
+            } catch (e: PasswordGeneratorExeption) {
+                Toast.makeText(requireActivity(), e.message, Toast.LENGTH_SHORT).show()
+                passwordText.text = ""
+            }
+            dialog.getButton(AlertDialog.BUTTON_NEUTRAL).setOnClickListener {
+                setPreferences()
+                try {
+                    passwordText.text = generate(callingActivity.applicationContext)[0]
+                } catch (e: PasswordGeneratorExeption) {
+                    Toast.makeText(requireActivity(), e.message, Toast.LENGTH_SHORT).show()
+                    passwordText.text = ""
+                }
+            }
+        }
+        return dialog
+    }
+
+    private fun setPreferences() {
```

Comment:
We can also pass our view here and remove all those not-null assertion operators.

Reply:
I find caching view lookups is quite error-prone because of configuration changes and such invalidating past references

Reply:
I was worried about multiple findViewById calls. So should we keep it same until we migrate to something like viewbinding ?

Reply:
They're fine for now. I think let's keep this until View Binding hits stable.

Reply:
Sure, I'll just make the pr for SshKeyGen and PasswordStore.